src/main/java/org/dice_research/SparqlUseCase/Query.java

```diff
@@ -0,0 +1,358 @@
+package org.dice_research.SparqlUseCase;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.BitSet;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.dice_research.vspace.Hypothesis;
+
+public class Query {
+	//original query
+	String query = null;
+	boolean mostGeneral = false;
+	
+	Map<String, String> prefixes;
+	List<Statement> statements;
+	List<Triple> triples;
+	int j=0;
```

Comment:
It is not a local variable since the parser uses it first to rename the variables found in the SELECT statement, and afterwards the query itself uses it to rename the variables in its body, so it is mainly used to keep track  of where the parser left of.
Might change this this if time allows it.

Reply:
1. To me, it seems like the variable is only used within the `renameVariables` method. So I do not really see a reason why it needs to be an attribute.
2. In case it should be an attribute, you should give it a meaningful name and add a javadoc comment to document its usage.

Reply:
Removed it entirely. Although had to change renameVariable's signature to include an integer which instructs the function to continue the renaming from the value where the parser left of.